### DISCUSSION TITLE
Getting started

### DISCUSSION THREAD

**Discussion by Haaselh0ff** [danielfernau/unifi-protect-video-downloader#304] [Q&A] [answered]
Is there a better guide on how to get started with this? I cannot for the life of me figure out how to actually try setting this up and using it. Sorry if this is a dumb question, I'm not very experienced with most of Docker or Python use.

Thanks for any help.

**Comment by danielfernau** [comment, marked as answer]
Hi there and thanks for your interest in the tool! 
I didn't have the time yet to write a full documentation on how to set things up, at least not in one place. So here's a quick summary of what you need to do:
1. Install Docker: https://docs.docker.com/get-docker/ (the Docker documentation page is a great place to get familiar with how it works and how you can use it)
2. Then you need to configure a local user on your UDMP/CK2+/UNVR/... by navigating to your console's IP address (or via unifi.ui.com) to access the management interface
3. There you need to select the "Admins" tab in the sidebar and click on "Add New Admin" on the page that opens
4. As `Role` select `Viewer`, then uncheck the `Allow Remote Access` box.
5. Choose a username and password for your new local user and write it down somewhere; you'll need it later on in the process
6. Click on `Add` to create the local user
7. Now open a Terminal (Linux or macOS) or a PowerShell window (Windows) and follow the instructions from the README page: https://github.com/danielfernau/unifi-protect-video-downloader#arrow_right-getting-started
8. For a full overview of options run the tool with `--help` or refer to https://github.com/danielfernau/unifi-protect-video-downloader/wiki/Usage-(v2.x)

If you get stuck along the way, just let me know and I'll try to provide additional guidance.

**Reply by Haaselh0ff** [reply]
Thank you for the assistance! I know you basically just said what was in the README but it made me retry and I got it all working! Only issue is Im not actually sure where it's downloading this footage to? I am on MacOS and the folder it's set to export to should be /downloads/ but am having no luck finding that when searching for it. 

Edit : Correction, I can find the folder that's create in the container in Docker and see that there is a large file (the video obviously) in it but I cannot view the file and nothing shows up when I click on the folder or try to export it.

Thanks again for the help!

**Reply by danielfernau** [reply]
You're welcome! Glad you could get it working. 

Yes, you have to enable file sharing in the Docker Desktop settings for it to be able to map your macOS folders to directories inside the running container. You can have a look at the documentation here on how to set it up: https://docs.docker.com/desktop/settings/mac/#file-sharing

The folder at the end of the command is the folder within the Docker container, and the volume mapping you specify as an option in the beginning is the macOS to container directory connection. 

Hope that helps a bit; let me know if you need any further assistance!

**Reply by Haaselh0ff** [reply]
Seems to be working very well for me now! Thank you for all the help! 

The only thing I'd like to know left is if there is a way to export footage as Unifi's Timelapse export so I can get that 600x footage without having to either use Protect's interface or without having to manually speed up footage. (If not that'd be a great feature to have!)

**Reply by danielfernau** [reply]
That's definitely something I'll look into. I'm pretty sure there's an API call for Timelapse export somewhere... Tracked as feature request in issue #308 🚀
